src/main/java/com/exasol/adapter/document/DocumentAdapter.java

```diff
@@ -198,4 +210,39 @@ public final SetPropertiesResponse setProperties(final ExaMetadata exaMetadata,
      * @return name of the database-specific adapter
      */
     protected abstract String getAdapterName();
+
+    protected abstract Capabilities getCapabilities();
+
+    @Override
+    public final GetCapabilitiesResponse getCapabilities(final ExaMetadata metadata,
+            final GetCapabilitiesRequest request) throws AdapterException {
+        final Capabilities capabilities = getCapabilities();
+        checkThatCapabilitiesAreSupported(capabilities.getMainCapabilities(), SUPPORTED_MAIN_CAPABILITIES, "main");
+        checkThatCapabilitiesAreSupported(capabilities.getPredicateCapabilities(), SUPPORTED_PREDICATE_CAPABILITIES,
+                "predicate");
+        checkThatCapabilitiesAreSupported(capabilities.getLiteralCapabilities(), SUPPORTED_LITERAL_CAPABILITIES,
+                "literal");
+        checkThatCapabilitiesAreSupported(capabilities.getAggregateFunctionCapabilities(),
+                SUPPORTED_AGGREGATE_FUNCTION_CAPABILITIES, "aggregate-function");
+        checkThatCapabilitiesAreSupported(capabilities.getScalarFunctionCapabilities(),
+                SUPPORTED_SCALA_FUNCTION_CAPABILITIES, "scala-function");
+        return GetCapabilitiesResponse.builder().capabilities(capabilities).build();
+    }
+
+    private void checkThatCapabilitiesAreSupported(final Set<? extends Enum> actualCapabilities,
+            final Set<? extends Enum> supportedCapabilities, final String capabilityType) {
+        final List<Enum> unsupportedCapabilities = actualCapabilities.stream()
+                .filter(mainCapability -> !supportedCapabilities.contains(mainCapability)).collect(Collectors.toList());
+        if (!unsupportedCapabilities.isEmpty()) {
+            final String listOfUnsupported = unsupportedCapabilities.stream().map(Enum::toString)
+                    .sorted(String::compareTo).collect(Collectors.joining(", "));
+            final String listOfSupported = supportedCapabilities.stream().map(Enum::toString).sorted(String::compareTo)
+                    .collect(Collectors.joining(", "));
+            throw new UnsupportedOperationException(
+                    "F-VSD-3 This dialect specified " + capabilityType + "-capabilities (" + listOfUnsupported
```

Comment:
Do we agreed to follow error codes with colon `:`? Or it was decided?

Reply:
@AnastasiiaSergienko does it like that. Not sure if we have a decision on that. But the builder API will standardize it anyway ;-)